source/SH3/arc/vfile.cpp

```diff
@@ -31,5 +57,16 @@ std::size_t sh3_arc_vfile::ReadData(void* destination, std::size_t len, read_err
         e.set_error(read_result::PARTIAL_READ);
     }
 
-    std::memcpy(destination, buffer.data() + fpos, len);
+    // Prevent memcpy from reading past the vector
+    if(fpos + len > fsize)
+    {
+        std::size_t nbytes = len - fpos
+        std::memcpy(destination, buffer.data() + fpos, nbytes);
+    }
+    else
+    {
+        std::memcpy(destination, buffer.data() + fpos, len);
```

Comment:
This almost duplicated line isn't needed if `std::memcpy` is always called with `nbytes`, which is set to the appropriate size.